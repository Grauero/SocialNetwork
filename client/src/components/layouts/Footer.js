import React from 'react';

const Footer = () => (
  <footer className="bg-light text-dark mt-5 p-4 text-center">
    Copyright &copy; {new Date().getFullYear()} Network
  </footer>
);

export default Footer;
