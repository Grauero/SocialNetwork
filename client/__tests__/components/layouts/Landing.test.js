import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { Landing } from '../../../src/components/layouts/Landing';

const props = {
  auth: { isAuthenticated: true },
  history: { push: jest.fn() }
};
const store = createStore(() => {});
let wrapper, component;

beforeEach(() => {
  wrapper = mount(
    <BrowserRouter>
      <Provider store={store}>
        <Landing {...props} />
      </Provider>
    </BrowserRouter>
  );
  component = wrapper.find(Landing);
});

afterEach(() => wrapper.unmount());

it('redirects to /dashboard if user is authenticated', () => {
  expect(props.history.push).toHaveBeenCalled();
  expect(props.history.push).toHaveBeenCalledTimes(1);
  expect(props.history.push).toHaveBeenCalledWith('/dashboard');
});

it('renders markup if user isnt authenticated', () => {
  expect(toJSON(component)).toMatchSnapshot();
});
