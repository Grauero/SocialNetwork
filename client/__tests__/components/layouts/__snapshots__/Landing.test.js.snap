// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders markup if user isnt authenticated 1`] = `
<Landing
  auth={
    Object {
      "isAuthenticated": true,
    }
  }
  history={
    Object {
      "push": [Function],
    }
  }
>
  <div
    className="landing"
  >
    <div
      className="dark-overlay landing-inner text-light"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col-md-12 text-center"
          >
            <h1
              className="display-3 mb-4"
            >
              Developer Network
            </h1>
            <p
              className="lead"
            >
              Create a developer profile/portfolio, message other people, share posts and exchange views with other developers
            </p>
            <hr />
            <Link
              className="btn btn-lg btn-light mr-2 signup"
              replace={false}
              to="/register"
            >
              <a
                className="btn btn-lg btn-light mr-2 signup"
                href="/register"
                onClick={[Function]}
              >
                Sign Up
              </a>
            </Link>
            <Link
              className="btn btn-lg btn-light"
              replace={false}
              to="/login"
            >
              <a
                className="btn btn-lg btn-light"
                href="/login"
                onClick={[Function]}
              >
                Login
              </a>
            </Link>
          </div>
        </div>
      </div>
    </div>
  </div>
</Landing>
`;
